src/pest_parser/hash.pest

```diff
@@ -0,0 +1,429 @@
+spread_op       = {"..."}
+dot_op          = {"."}
+semi            = {";"}
+comma           = {","}
+
+bit_shift_op = _{shl_op | shr_op}
+bit_mod_op   = _{andb_op | orb_op | xorb_op}
+logical_op   = _{andl_op | orl_op}
+eqneq_op     = _{ref_eq_op | comp_op | neq_op}
+muldiv_op    = _{mul_op | div_op}
+addsub_op    = _{add_op | sub_op}
+
+/// Since we use the '<' & '>' as syntax for type arguments, the grammars
+/// for the mathematical relations are called *_angle_br
+relational_op = _{leq_op | geq_op | left_angle_br | right_angle_br}
+
+assignment_operator = _{
+    eq_op           |
+    increment_op    |  
+    decrement_op    |
+    mul_inplace_op  |
+    div_inplace_op  |
+    mod_inplace_op  |
+    andl_inplace_op |
+    orl_inplace_op  |
+    andb_inplace_op |
+    orb_inplace_op  |
+    xor_inplace_op
+}
+
+unary_operator = _{
+    not_op  |
+    notb_op |
+    add_op  |
+    sub_op
+}
+
+binary_operator = _{
+    eqneq_op     |
+    as_op        |
+    muldiv_op    |
+    addsub_op    |
+    logical_op   |
+    bit_shift_op |
+    bit_mod_op   |
+    relational_op
+}
+
+/// operators map
+operators = {
+    "===" | "==" | "=" |
+    "=>" | "<=" | ">=" |
+    "&&" | "||" |
+    "&&=" | "||=" |
+    "+=" | "-=" | "*=" | "/=" | "%=" | "!=" |
+    "+" | "-" | "*" | "/" | "%" | "!" | "**" |
+    "&=" | "^=" | "|=" |
+    "&" | "^" | "|" |
+    "?" | "::" | ":" | "..." | "." | ";" | ","
+}
+
+/// Individual operator parsers
+as_op = {"as"}
+
+
+/// Keyword map and keyword parsing
+keywords = {
+    "if" | "else" | "match" | 
+    "let" | "struct" | "enum" | "trait" |
+    "loop" | "while" | "for" | "in" |
+    "break" | "continue" | "return" |
+    "import" | 
+    "as" | "where"
+}
+
+/// Brackets map
+bracket = {
+    "(" | ")" | "{" | "}" |
+    "[" | "]" | "<" | ">"
+}
+
+// Specific bracket matchers
+left_angle_br  = { "<" }
+right_angle_br = { ">" }
+
+
+/// Type parsers that annotate the types of more complicated types such as 
+/// set, map, tuple, functions, etc.
+set_type         = _{"{" ~ type_t ~ "}"}
+map_type         = _{"{" ~ type_t ~ ":" ~ type_t ~ "}"}
+list_type        = _{"[" ~ type_t ~ "]"}
+tuple_type       = _{"(" ~ (type_t ~ ("," ~ type_t)*) | (",") ~ ")"}
+function_type    = _{"(" ~ (type_t ~ (comma ~ type_t)*)? ~ ")" ~ "=>" ~ type_t}
+existential_type = _{"?"}
+infer_type       = _{"_"}
+
+type_var = {'A'..'Z'}
+
+named_type = {
+    infer_type |
+    type_var   |
+    access_name ~ type_args?
+
+}
+
+/// Name which might be nested within namespaces (e.g. a::b::c) where 'a', 'b' are
+/// namespaces, and 'c' is the actual name.
+name        = {ident}
+access_name = {name ~ ("::" ~ name)*}
+
+/// Types
+type_t = { 
+    function_type    |
+    tuple_type       |
+    list_type        |
+    set_type         |
+    map_type         |
+    existential_type |
+    named_type
+ }
+
+/// Import statement
+import = {"import" ~ "(" ~ string ~ ")"} // @@Improvement: in the future change this to an expression
```

Comment:
Why change it to an expression?

Reply:
So that it in the future we may want to support dynamic imports